packages/remix-dev/config/flat-routes.ts

```diff
@@ -103,8 +103,11 @@ export function flatRoutesUniversal(
   return routes;
 }
 
-function isIndexRoute(routeId: string) {
-  return routeId.endsWith("_index");
+export function isIndexRoute(routeId: string) {
+  let isFlatFile = !routeId.includes(path.posix.sep);
+  return isFlatFile
+    ? routeId.endsWith("_index")
+    : /^(.*)\/index$/.test(routeId);
```

Comment:
nit: I don't think the `^(.*)` is necessary

```suggestion
    : /\/index$/.test(routeId);
```

Reply:
yeah i had it like that originally, but thought better safe than sorry 🤷‍♂️